### PR TITLE
change: [M3-6541] - eslint: consistent-type-imports

### PR DESCRIPTION
**Pull request by abailly-akamai**
## Description 📝
Tiny PR to start enforcing `consistent-type-imports` through the codebase (`import type` for type file)

https://typescript-eslint.io/rules/consistent-type-imports

This will remain a warning it does not affect the developer experience negatively other than suggesting to make that change when creating or editing a file. It will also make it easier/quicker to fix those via shortcuts when encountered.

## Changes  🔄
- Add `consistent-type-imports` warning to our `es-lint` rules

## Preview 📷
**Include a screenshot or screen recording of the change**

Screenshot 2024-06-03 at 12 39 07

### Verification steps
- Pull locally and open random file to see the new lint warning

## As an Author I have considered 🤔

*Check all that apply*

- [ ] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [x] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [x] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

**Comment by bnussman-akamai** [comment]
Curious to hear your thoughts on https://github.com/microsoft/TypeScript/issues/39861#issuecomment-668131921

**Comment by abailly-akamai** [comment]
@bnussman-akamai I agree it's mostly stylistic at this point (i was convinced there was another element to it, but maybe not or irrelevant to our bundling/compiling setup). 

I think that comment is obsolete however, since

> and since there are currently no tools that would enforce this style, it would fall on you to analyze and separate your declarations manually, wasting your valuable coding time.

isn't true anymore.

Thing is, we have agreed and started to implement this style (and people usually point to it in code reviews), so this PR is for consistency sake. Not a hill i would die on, I was again mostly going for consistency here. I don't even know if we can go back to grouping imports via linting, so our options are: we leave things as is and we don't think about it and stop commenting on PRs _OR_ we go with the lint rule.

**Review by bnussman-akamai — APPROVED** [review]
New rules works as expected ✅

File: packages/manager/.changeset/pr-10540-added-1717433580405.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@linode/manager": Added
+---
+
+New `consistent-type-imports` es-lint warning ([#10540](https://github.com/linode/manager/pull/10540))
```
Comment (mjac0bs) [inline packages/manager/.changeset/pr-10540-added-1717433580405.md:5]: Not customer facing, so this may be better under Tech Stories.

**Review by mjac0bs — APPROVED** [review]
Thanks for improving the developer experience - confirmed the linter is flagging these instances as a warning.

Screenshot 2024-06-05 at 10 49 20 AM

**Pull request merged**